graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/fetchers/deployed/MutationPayload.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.graphql.schema.graphqlfirst.fetchers.deployed;
+
+import io.stargate.auth.TypedKeyValue;
+import io.stargate.db.query.BoundQuery;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * Abstract the queries needed to process a mutation operation, and how the results will be turned
+ * into a GraphQL response.
+ */
+class MutationPayload<ResultT> {
```

Comment:
Does this need to be generic? As far as I can tell, all the parameterized inheritors use `DataFetcherResult<Object>`.

Reply:
Fair point. And this also applies to `MutationFetcher` itself.

`ResultT` made sense in the beginning because the mutations had specific return types. But as we added more ways to map results, `Object` is convenient for flexibility.

I'll make the change.

Reply:
Actually, there's one argument to keep the code generic: I think the CQL-first fetchers could be retrofitted to use it too, and they have more specific return types.